tests/app/main/test_suppliers.py

```diff
@@ -182,24 +169,30 @@ def test_shows_edit_buttons(self, get_current_suppliers_users, data_api_client):
             document = html.fromstring(res.get_data(as_text=True))
 
             assert document.xpath(
-                "//a[normalize-space(string())=$t][@href=$u][contains(@class, $c)]",
-                t="Edit",
-                u="/suppliers/edit",
-                c="summary-change-link",
-            )
-
-            assert document.xpath(
-                "//tr[./td[normalize-space(string())=$f]][.//a[normalize-space(string())=$t][@href=$u]]",
+                "//*[(.//h3)[1][normalize-space(string())=$f]][.//a[normalize-space(string())=$t][@href=$u]]",
                 f="G-Cloud 6",
                 t="View services",
                 u="/suppliers/frameworks/g-cloud-6/services",
             )
+
             assert not document.xpath(
-                "//tr[./td[normalize-space(string())=$f]][.//a[normalize-space(string())=$t]]",
+                "//*[(.//h3)[1][normalize-space(string())=$f]][.//a[normalize-space(string())=$t]]",
                 f="G-Cloud 7",
                 t="View services",
             )
             assert not document.xpath("//a[@href=$u]", u="/suppliers/frameworks/g-cloud-7/services")
+            assert not document.xpath(
```

Comment:
Although I think in this case I wouldn't want to go as far as just checking the framework name isn't in the output because I wouldn't want to rule out that there could be some line that says "Hey <framework_name> is coming" or "<framework_name> has ended" - if anyone ever added such a section they would find it hitting this assertion and almost certainly say "that doesn't apply anymore" and delete it, leaving us without this assertion.

 * [ ] But I can certainly slacken it.